scripts/generate_BERTMap_predictions.py

```diff
@@ -0,0 +1,495 @@
+        [
+            target_ontology_train,
+            source_ontology_train,
+            *source_ontologies_inference,
+            *target_ontologies_inference,
+        ],
+        desc="Downloading ontologies",
+        leave=False,
+    ):
+        ext = ".ttl" if prefix.upper() == "MESH2024" else ".owl"
+        ontology_path = ONTOLOGIES_DIRECTORY.joinpath(prefix.lower()).with_suffix(ext)
+        prefix_to_path[prefix.lower()] = ontology_path
+        if not ontology_path.is_file():
+            tqdm.write(f"Downloading {prefix}")
+            download(url=PREFIX_TO_DOWNLOAD_URL[prefix.upper()], path=ontology_path)
+    return prefix_to_path
+
+
+def load_bertmap(
+    config_path: Path,
+    target_ontology_train: str,
+    source_ontology_train: str,
+    ontology_paths: dict[str, Path],
+    use_biomappings: bool = False,
+    train_model: bool = False,
+) -> BERTMapPipeline:
+    """Load in the bertmap model (will download from hugging face if not present in ./bertmap)."""
+    config = BERTMapPipeline.load_bertmap_config(config_path.as_posix())
+
+    if use_biomappings:
+        print("-" * 100)
+        print("generating known mappings")
+        config.known_mappings = save_known_maps(
+            target_ontology_train=target_ontology_train,
+            source_ontology_train=source_ontology_train,
+            mappings_directory=BERTMAP_DIR.joinpath("knownMaps"),
+        ).as_posix()
+    print("-" * 100)
+    print(f"Using config: \n{config}")
+    print("-" * 100)
+    if not train_model:
+        config.global_matching.enabled = False
+        if not os.path.isdir("bertmap"):
+            print("downloading model from hugging face")
+            snapshot_download(repo_id="buzgalbraith/BERTMAP-BioMappings", local_dir="./")
+        else:
+            print("Model found at bertmap")
+    src_onto = Ontology(ontology_paths[source_ontology_train.lower()].as_posix())
+    target_onto = Ontology(ontology_paths[target_ontology_train.lower()].as_posix())
+    return BERTMapPipeline(src_onto, target_onto, config)
+
+
+def save_known_maps(
+    target_ontology_train: str, source_ontology_train: str, mappings_directory: Path
+) -> Path:
+    """Use true mappings from Biomappings as ground truth for training BertMap Model."""
+    true_mappings = biomappings.load_mappings()
+
+    # onto_filter = lambda x: (x["target prefix"].lower() == target_ontology_train.lower()) & (
+    #     x["source prefix"].lower() == source_ontology_train.lower()
+    # )
+    # iri_map = lambda x: (
+    #     SOURCE_PREFIX_IRI_MAPS[source_ontology_train.lower()](x["source identifier"])
+    #     + "\t"
+    #     + SOURCE_PREFIX_IRI_MAPS[target_ontology_train.lower()](x["target identifier"])
+    #     + "\t1.0\n"
+    # )
+    def onto_filter(x):
+        return (x["target prefix"].lower() == target_ontology_train.lower()) & (
+            x["source prefix"].lower() == source_ontology_train.lower()
+        )
+
+    def iri_map(x):
```

Comment:
these functions don't make sense - you could accomplish the same thing below in a list comprehension that would be much more readable and understandable